### DISCUSSION TITLE
schema validation warnings during mvn oc:resource and gradle k8sResource

### DISCUSSION THREAD

**Discussion by sockejr** [eclipse/jkube#1932] [Q&A] [answered]
Hello in my mvn builds in the goal oc:resource I get the following warnings:
`
Unknown keyword $module - you should define your own Meta Schema. If the keyword is irrelevant for validation, just use a NonValidationKeyword
Unknown keyword existingJavaType - you should define your own Meta Schema. If the keyword is irrelevant for validation, just use a NonValidationKeyword
Unknown keyword javaOmitEmpty - you should define your own Meta Schema. If the keyword is irrelevant for validation, just use a NonValidationKeyword
`
I have also seen this error in the quickstart examples e.g. https://github.com/eclipse/jkube/blob/master/quickstarts/gradle/webapp-jetty/README.md and others. So gradle builds are also affected.

I have found out so far that the error message is coming from the json validation in library networknt / json-schema-validator
probably called in class org.eclipse.jkube.kit.common.util.validator.ResourceValidator.

Some hints I have found in this discussion: https://github.com/networknt/json-schema-validator/issues/79

Is it the right way to just suppress the warning by adding the nonValidationKeyword ?

**Comment by manusa** [comment, marked as answer]
This is something that we should have fixed long ago.

> Is it the right way to just suppress the warning by adding the nonValidationKeyword ?

I haven't looked into it, but it probably is the way.

**Reply by sockejr** [reply]
bug report opened: https://github.com/eclipse/jkube/issues/1934
